dbv2/reader.go

```diff
@@ -0,0 +1,189 @@
+package dbv2
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var (
+	readerBufferSize = validLength * (bufferSize / 10)
+	// readerOffsetBytes represents the number of bytes that occupy the meta-data section of the table. This can be discarded
+	// during parse operation as this is only to know the type of the table.
+	readerOffsetBytes = len([]byte(`maxValidLength: 100 chars
+
+`))
+)
+
+type Reader interface {
+	Read()
+}
+
+type DataReader struct {
+	dataTable *DataTable
+	reader    *bufio.Reader
+}
+
+func NewDataReader(path string) (*DataReader, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("new data-reader: %w", err)
+	}
+	return &DataReader{
+		dataTable: &DataTable{
+			File: f,
+		},
+		reader: bufio.NewReaderSize(f, readerBufferSize),
+	}, nil
+}
+
+// discardInitialOffset discards the initial meta-data of the table file.
+func (dr *DataReader) discardInitialOffset() error {
+	discarded, err := dr.reader.Discard(readerOffsetBytes)
+	if err != nil {
+		return fmt.Errorf("discard initial offset: %w", err)
```

Comment:
Need a way to panic here and send a 500 response don't you think?

Reply:
No, we should never panic. That will stop the flow of Bench-routes. Systems should be recoverable. Panicking makes them fatal and hence non-recoverable. Errors should always be traced.